physicalplan/aggregate/hashaggregate.go

```diff
@@ -140,7 +176,7 @@ func (a *aggregate) workerTask(workerID int, vector model.StepVector) model.Step
 }
 
 func (a *aggregate) initializeVectorizedTables() ([]aggregateTable, []labels.Labels, error) {
-	tables, err := newVectorizedTables(a.stepsBatch, a.aggregation)
+	tables, err := newVectorizedTables(a.stepsBatch, a.aggregation, a.param)
```

Comment:
Unfortunately we have to do validation when we create the operator. If we throw validation errors during execution, we cannot fall back to the old engine for unsupported operations.

Reply:
If the aggregation param validation fails, do we want to fallback to the old engine? I think they will also fail, too

Reply:
What I meant with validation is validating whether the engine has support for operators in the promql query. In case of no support, we don't want to fail mid-execution because that prevents us from using the fallback. Even if we did have a mechanism to fall back in that case, we would have wasted lots of time and made unnecessary network calls.

In case of actual PromQL validation issues, I think it should be safe to fail whenever we see fit. One example is many-to-many join which is detected at query execution time. The old engine would fail as well, so no reason to fall back as you indicated.

Reply:
I will try to unify the both interfaces so that we can validated early in execution call.